### ISSUE TITLE
brew search pkgName errors: undefined method `path' for nil:NilClass

### ISSUE THREAD

**Issue by todd-dsm** [Homebrew/brew#2463]
```bash
brew search /packer/
packer ✔     packer-completion ✔
Error: undefined method `path' for nil:NilClass
Please report this bug:
  http://docs.brew.sh/Troubleshooting.html
/usr/local/Homebrew/Library/Homebrew/missing_formula.rb:125:in `deleted_reason'
/usr/local/Homebrew/Library/Homebrew/missing_formula.rb:10:in `reason'
/usr/local/Homebrew/Library/Homebrew/cmd/search.rb:70:in `search'
/usr/local/Homebrew/Library/Homebrew/brew.rb:91:in `<main>'
```

The only thing that is different is today I accepted the install for the macOS 10.12.4 update. To balance though, I don't search for new stuff to install every day.

# Troubleshooting Steps:
Ran `brew update` twice.

```bash
brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry and just ignore them. Thanks!

Warning: Putting non-prefixed coreutils in your path can cause gmp builds to fail.

Warning: Putting non-prefixed findutils in your path can cause python builds to fail.

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/python2.7/greenlet/greenlet.h
  /usr/local/include/python3.6m/greenlet/greenlet.h
```

```bash
brew config
HOMEBREW_VERSION: 1.1.12
ORIGIN: https://github.com/Homebrew/brew
HEAD: 8ea778f7c7ea790694485030d105a92a9416ef33
Last commit: 5 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 8f27ebcb7e82f8be5e500fa7eac2e7dac2312b7b
Core tap last commit: 9 minutes ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: octa-core 64-bit haswell
Homebrew Ruby: 2.0.0-p648
Clang: 8.1 build 802
Git: 2.12.2 => /usr/local/bin/git
Perl: /usr/bin/perl
Python: /usr/local/bin/python => /usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/bin/python2.7
Ruby: /usr/local/bin/ruby => /usr/local/Cellar/ruby/2.4.1_1/bin/ruby
Java: 1.8.0_121
macOS: 10.12.4-x86_64
Xcode: N/A
CLT: 8.3.0.0.1.1490382677
X11: N/A
```

**Comment by MikeMcQuaid** [comment]
Fixed by 171f65a1e44b4def783f67e1a0ac19d65ed495d9

**Comment by todd-dsm** [comment]
Excellent, thank you @MikeMcQuaid; the new results:

```bash
$ brew search /packer/
packer ✔         packer-completion ✔
Caskroom/cask/texturepacker
```
